src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/KeyStatisticServiceTests.cs

```diff
@@ -1235,64 +1219,73 @@ await using (var context = InMemoryContentDbContext(contextId))
         {
             var keyStatisticService = SetupKeyStatisticService(context);
             var result = await keyStatisticService.Delete(
-                Guid.NewGuid(),
-                keyStatisticDataBlock.Id);
+                release.Id,
+                Guid.NewGuid());
 
             result.AssertNotFound();
         }
     }
 
     [Fact]
-    public async Task Delete_NoKeyStatistic()
+    public async Task Delete_KeyStatAttachedToDifferentRelease()
     {
         var releaseId = Guid.NewGuid();
-        var dataBlockId = Guid.NewGuid();
+        var otherReleaseId = Guid.NewGuid();
 
-        var keyStatisticDataBlock = new KeyStatisticDataBlock
+        var keyStatisticText = new KeyStatisticText
         {
-            ReleaseId = releaseId,
-            DataBlockId = dataBlockId,
+            Title = "title",
+            Statistic = "Over 9000",
+            ReleaseId = otherReleaseId,
             Trend = "trend",
             GuidanceTitle = "guidanceTitle",
             GuidanceText = "guidanceText",
             Created = new DateTime(2000, 1, 1),
             Updated = null,
         };
 
-        var release = new Release
+        var incorrectRelease = new Release
         {
             Id = releaseId,
-            ContentBlocks = new List<ReleaseContentBlock>
-            {
-                new()
-                {
-                    ContentBlock = new DataBlock { Id = dataBlockId },
-                },
-            },
             KeyStatistics = new List<KeyStatistic>
             {
-                keyStatisticDataBlock,
                 new KeyStatisticText(),
                 new KeyStatisticText(),
+                new KeyStatisticDataBlock(),
             }
         };
 
+        var release = new Release
+        {
+            Id = otherReleaseId,
+            KeyStatistics = new List<KeyStatistic>
+            {
+                keyStatisticText
+            },
+        };
+
         var contextId = Guid.NewGuid().ToString();
         await using (var context = InMemoryContentDbContext(contextId))
         {
-            await context.Releases.AddRangeAsync(release);
+            await context.Releases.AddRangeAsync(incorrectRelease, release);
             await context.SaveChangesAsync();
         }
 
         await using (var context = InMemoryContentDbContext(contextId))
         {
             var keyStatisticService = SetupKeyStatisticService(context);
             var result = await keyStatisticService.Delete(
-                release.Id,
-                Guid.NewGuid());
+                incorrectRelease.Id,
+                keyStatisticText.Id);
 
             result.AssertNotFound();
         }
+
+        await using (var context = InMemoryContentDbContext(contextId))
+        {
+            var keyStatistics = context.KeyStatistics.ToList();
+            Assert.Equal(4, keyStatistics.Count);
+        }
     }
```

Comment:
I think in general the tests in this class are quite bloated with properties and extra data that aren't really required.

I'm not too bothered if you don't want to make this change, but I'll give an example for this particular unit test of how short this could be avoiding setting up all the id's, extra properties, and avoiding the collection of key stats on the releases.

```
var release = new Release();
var otherRelease = new Release();

var keyStatisticText = new KeyStatisticText
{
  Release = release,
  ...<any other attributes that are really necessary>
};


Save the release, otherRelease, and the keyStatisticText.

Call the service delete method with keyStatisticText.Id and otherRelease.Id

Check for the not found error.

Check that the one key stat still exists.
```

Reply:
I've changed the one test you highlighted. I think you're right it could have been more succinct. I've left the rest of the tests, but will try to do this better from now on.